Blog post on matching nodes and specifiers

gaurav opened the pull request:
I wrote a blog post about how we plan to match specifiers with nodes in phylogenies, updated to the discussion with Hilmar on Tuesday. It's changed quite a bit from when Guanyang saw it last week, so let me know if I've gotten something completely wrong!

gaurav commented:
@ncellinese @GZhang2 This blog post is pretty specific about how we could construct specifiers and taxonomic units in OWL and use them to link specifiers to nodes, so I'm not going to request reviews from you. But please have a look at it now or after it's published and let me know what you think!

ncellinese commented:
I can look at this as soon as I can given I am swamped right now, but are we going to write a blog for every discussion we engage into?  Can we actually do some serious hacking instead?

Nico




> On Dec 7, 2017, at 4:43 PM, Gaurav Vaidya <notifications@github.com> wrote:
> 
> @ncellinese <https://github.com/ncellinese> @GZhang2 <https://github.com/gzhang2> This blog post is pretty specific about how we could construct specifiers and taxonomic units in OWL and use them to link specifiers to nodes, so I'm not going to request reviews from you. But please have a look at it now or after it's published and let me know what you think!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/phyloref/phyloref.github.io/pull/28#issuecomment-350103350>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACaXwZBV6XaUbmeHgThbVMRhswW2UKZ4ks5s-FvygaJpZM4Q6MP9>.
>

hlapp commented:
@ncellinese I find it pretty critical that we write down the decisions we are going to implement, or otherwise we're just going to go in circles or end up with hacks that nobody agrees what they should or shouldn't be doing.

ncellinese commented:
Well, I just think blogging constantly is very time consuming.


> On Dec 7, 2017, at 5:23 PM, Hilmar Lapp <notifications@github.com> wrote:
> 
> @ncellinese <https://github.com/ncellinese> I find it pretty critical that we write down the decisions we are going to implement, or otherwise we're just going to go in circles or end up with hacks that nobody agrees what they should or shouldn't be doing.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/phyloref/phyloref.github.io/pull/28#issuecomment-350112923>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACaXwfSlTjvgi4Swm8-VlyaTGcn1o419ks5s-GVUgaJpZM4Q6MP9>.
>

hlapp commented:
This isn't blogging as much as it is documenting what we decided to do at some point and why. We'd need to do it anyway or within soon we can't explain anymore why we are doing things the way we are.

ncellinese commented:
Ok, ok. I read it and I have no conceptual problems with the blog.  

Nico




> On Dec 7, 2017, at 6:20 PM, Hilmar Lapp <notifications@github.com> wrote:
> 
> This isn't blogging as much as it is documenting what we decided to do at some point and why. We'd need to do it anyway or within soon we can't explain anymore why we are doing things the way we are.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/phyloref/phyloref.github.io/pull/28#issuecomment-350124997>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACaXwRNMlOqpM873_dhV84O4esx40Jyvks5s-HK9gaJpZM4Q6MP9>.
>

gaurav commented:
@hlapp Okay, this is finally ready for you to look at again! I changed the model described in the first half of the blog post to bring it in line with [the latest iteration of the curation workflow](https://github.com/phyloref/curation-workflow/pull/7) and cleaned up the text a bit. The changes to the second half of the blog post are much more substantial, moving away from talking specifically about the Phylocode to talking about matching taxonomic units more generally. Of course, we can always [undo this revision](https://github.com/phyloref/phyloref.github.io/pull/28/commits/dc957d0ee689d23cb868b89256ec3672fd0ce2dd) if you think I headed in the wrong direction. Let me know what you think!

hlapp commented:
I have a couple of minor edits but overall good job and in good shape. My one major comment that I think needs to be addressed in some way is that in the property-centric model there are now multiple TUs associated with a Node instance, and whether these are the same (unlikely) or how they differ is confusing, including how this does or does not likely mesh with the authors original intent. In truth, due to the OWA there is always implicit the possibility of other TU instance associations that just haven’t been asserted, so the model isnt asserting something that would otherwise be known not to exist. But I think this deserves a few sentences.

gaurav commented on _posts/blog/2016-10-05-postdoc2-ad.md:
```diff
@@ -99,6 +99,6 @@ filled.
 
 [Phyloreferencing project]: http://phyloref.org
-[existing postdoc]: {% post_url /blog/2016-07-20-welcoming-gaurav %}
+[existing postdoc]: {% post_url 2016-07-20-welcoming-gaurav %}
```
Jekyll wouldn't run unless all of the `post_url` paths resolve correctly. When I installed it on either my work or home machine, it wouldn't run until I "fixed" this, which I did. I later got a warning in the opposite direction, telling me to undo this "fix". This time, Jekyll did work, so I decided to leave it as such rather than adding another commit to undo a previous, apparently unnecessary fix. [The Jekyll documentation](https://jekyllrb.com/docs/templates/#linking-to-posts) wasn't clear enough to make it clear to me which one was the correct form, so I went for the one that worked correctly.

I've just confirmed that the `post_url 2016-07-...` form *does* work, but it does so with a warning. However, `... /blog/...` does not give any error, so it looks like it is the correct form! I've corrected this back to `post_url /blog/2016-07-20-welcoming-gaurav`.

gaurav commented on _posts/blog/2017-05-05-welcoming-guanyang.md:
```diff
@@ -11,7 +11,7 @@ image:
 ---
 
-A few months ago we [opened the search]({% post_url /blog/2016-10-05-postdoc2-ad %}) for the second[^postdocs]
+A few months ago we [opened the search]({% post_url 2016-10-05-postdoc2-ad %}) for the second[^postdocs]
```
This link also works, also with a warning. Fixed to `{% post_url /blog/2016-10-05-postdoc2-ad %}`.

gaurav commented on _posts/blog/2017-05-05-welcoming-guanyang.md:
```diff
@@ -36,7 +36,7 @@ expanding his training. He will be particularly involved in
 
 Guanyang participated in the [face-to-face project meeting]({%
-post_url /blog/2017-04-26-first-face-to-face-project-meeting %}) last
+post_url blog/2017-04-26-first-face-to-face-project-meeting %}) last
```
This link *also* works, and this time without a warning! I've fixed it to `{% post_url /blog/2017-04-26-first-face-to-face-project-meeting %}`.

gaurav commented:
@hlapp I've added a paragraph about this, but I'm not sure I'm fully happy with this yet. What do you think?

hlapp commented on _posts/blog/2018-01-29-matching-nodes-to-phyloreferences.md:
```diff
@@ -15,20 +15,20 @@ image:
 Since developing a [system for phyloreferencing based on nodes in a particular phylogeny]({% post_url /blog/2017-03-21-a-phyloreferencing-experiment %}) earlier this year, we've been working on extending it to support node- and branch-based definitions — if you know how to read [pytest](https://www.pytest.org/) and [TAP](https://en.wikipedia.org/wiki/Test_Anything_Protocol) output, you can see th ...
 
-Nodes on a phylogeny may be identified to a specimen, or a taxon (whether species or higher level), and may also be associated with morphological or molecular character data. In the [Comparative Data Analysis Ontology](http://www.obofoundry.org/ontology/cdao.html) (CDAO, described in Prosdocimi *et al.*, 2009[^prosdocimi_et_al_2009]), this connection goes through an intermediate entity called a [ ...
+**Annotating nodes on a phylogeny.** Nodes on a phylogeny may be identified to a specimen or a taxon, whether at the species rank or higher, and may also be associated with morphological or molecular character data. In the [Comparative Data Analysis Ontology](http://www.obofoundry.org/ontology/cdao.html) (CDAO, described in Prosdocimi *et al.*, 2009[^prosdocimi_et_al_2009]), this connection goes  ...
```
_or a lineage-specific taxonomy ontologies_ -> remove 'a'

hlapp commented:
@gaurav note that the squashed merge commit (e996f367a401062e041a4f43e4c48aacd7e3ef72) only shows (as it should) the new blog post file and the included image as changed. So your note in the commit message about how it came to is therefore confusing (especially in the future).

My general recommendation for this kind of scenario, i.e., where a change is developed in an one area of the repo over a period of time while other areas experience concurrent but otherwise unrelated and non-conflicting changes is to rebase against master, repeatedly if changes just keep being made to master. There's nothing evil about rebasing on a branch, and in fact if done in the interest of keeping a clean change set against master it is one of the powerful features of git.

gaurav commented:
Will do! I did try a quick rebase before I squash-merged, but quickly ran into conflicts, and decided that since the squash-merge was going to wipe out the effect of those changes, I could just document it and move on. I'll be better about rebasing regularly in the future!
